### PR TITLE
Skip notifications when OCR mode is "skip file"

### LINKED ISSUE #232: Get feedback via Notifications - too much unuseful notifications
I set the "OCR mode" to "skip file completely":

image

When uploading a file which have OCR content already, I get a notification about "Workflow OCR error", but there is no error, no output is ok because OCR is not needed. 
In the log I see this:

```
[workflow_ocr] Fehler: OCA\WorkflowOcr\Exception\OcrNotPossibleException: OCRmyPDF did not produce any output for file /uwe/files/ocrPDF/filename.pdf at <<closure>>

0. /home/www/nextcloud/apps/workflow_ocr/lib/Service/OcrService.php line 60
   OCA\WorkflowOcr\OcrProcessors\OcrMyPdfBasedProcessor->ocrFile()
1. /home/www/nextcloud/apps/workflow_ocr/lib/BackgroundJobs/ProcessFileJob.php line 146
   OCA\WorkflowOcr\Service\OcrService->ocrFile()
2. /home/www/nextcloud/apps/workflow_ocr/lib/BackgroundJobs/ProcessFileJob.php line 108
   OCA\WorkflowOcr\BackgroundJobs\ProcessFileJob->processFile()
3. /home/www/nextcloud/lib/public/BackgroundJob/Job.php line 81
   OCA\WorkflowOcr\BackgroundJobs\ProcessFileJob->run()
4. /home/www/nextcloud/lib/public/BackgroundJob/QueuedJob.php line 57
   OCP\BackgroundJob\Job->start()
5. /home/www/nextcloud/lib/public/BackgroundJob/QueuedJob.php line 47
   OCP\BackgroundJob\QueuedJob->start()
6. /home/www/nextcloud/cron.php line 152
   OCP\BackgroundJob\QueuedJob->execute()
```

### PR DESCRIPTION
**Pull request by R0Wi**
* Fixes #232
* Move logic to OcrService

### CODE REVIEW

File: tests/Unit/Service/OcrServiceTest.php [hunk 1]
```diff
@@ -160,6 +237,339 @@ public function testCatchesTagNotFoundException() {
...
+		return $fileMock;
+	}
+
+	// public function dataProvider_ValidArguments() {
+	// 	$arr = [
+	// 		[['fileId' => 42, 'uid' => 'admin', 'settings' => '{}'], 'admin', '/admin/files', 'somefile.pdf', 'somefile.pdf'],
+	// 		[['fileId' => 42, 'uid' => 'myuser', 'settings' => '{}'], 'myuser', '/myuser/files', 'someotherfile.jpg', 'someotherfile.jpg.pdf']
+	// 	];
+	// 	return $arr;
+	// }
```
Comment (bahnwaerter) [inline tests/Unit/Service/OcrServiceTest.php:574]: Please remove unused code fragments.

```suggestion
```

File: lib/Service/IOcrService.php [hunk 2]
```diff
@@ -27,19 +27,19 @@
...
 
 interface IOcrService {
 	/**
-	 * Processes OCR on the given file
+	 * Processes OCR on the given file. Creates a new file version and emits appropriate events.
 	 *
-	 * @param string $mimeType        		The mimetype of the file to be processed
-	 * @param string $fileContent     		The file to be processed
+	 * @param int $fileId     				The id if the file to be processed
+	 * @param string $uid 					The id of the user who has access to this file
```
Comment (bahnwaerter) [inline lib/Service/IOcrService.php:36]: Please preserve indentation from existing parameter description below.

**Review by bahnwaerter — CHANGES_REQUESTED** [review]
Thanks @R0Wi for refactoring the OCR notification. The refactoring looks good to me. So feel free to merge your code changes after considering my two small comments.

**Comment by R0Wi** [comment]
Thanks for the heads up @bahnwaerter 👍

**Pull request merged**